### PR TITLE
Issue-30 ; Fixed matchesResourceIdName resulting in NotFoundException on Android P when trying expand WebView PopUp list

### PR DESCRIPTION
**Pull request by tpiDev**
Related issue: https://github.com/InflationX/Calligraphy/issues/30

### CODE REVIEW

File: calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java [hunk 1]
```diff
@@ -3,23 +3,27 @@
...
 import androidx.appcompat.widget.Toolbar;
 
 class Calligraphy {
 
     private static final String ACTION_BAR_TITLE = "action_bar_title";
     private static final String ACTION_BAR_SUBTITLE = "action_bar_subtitle";
+    private static final String TAG = Calligraphy.class.getSimpleName();
 
     /**
      * Some styles are in sub styles, such as actionBarTextStyle etc..
```
Comment (lucalves) [inline calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java:29]: ``` Suggestion: comment in only one line ```

File: calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java [hunk 2]
```diff
@@ -103,6 +107,16 @@ public Calligraphy(CalligraphyConfig calligraphyConfig) {
...
+        try {
+            return view.getResources().getResourceEntryName(view.getId());
+        } catch (Resources.NotFoundException exception) {
+            Log.w(TAG, "An error occurred when trying to get resource entry name. \nStackTrace:" + exception);
+            return "";
+        }
+    }
+
     /**
      * Handle the created view
```
Comment (lucalves) [inline calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java:121]: ``` Suggestion: comment in only one line ```

File: calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java [hunk 3]
```diff
@@ -103,6 +107,16 @@ public Calligraphy(CalligraphyConfig calligraphyConfig) {
         this.mAttributeId = new int[]{calligraphyConfig.getAttrId()};
     }
 
+    @NonNull
+    private static String getResourceEntryName(@NonNull View view) {
+        try {
+            return view.getResources().getResourceEntryName(view.getId());
+        } catch (Resources.NotFoundException exception) {
+            Log.w(TAG, "An error occurred when trying to get resource entry name. \nStackTrace:" + exception);
```
Comment (jbarr21) [inline calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java]: Let's remove the newline and stacktrace label and just pass the exception as the third method param

Comment (tpiDev) [reply calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java]: Done.

File: calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java (hunk 2 above)
Comment (tpiDev) [reply calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java:121]: @lucalves I am not sure why you suggesting this change here. Javadoc has not changed and it's not part of this pull request. Won't change.

File: calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java (hunk 1 above)
Comment (tpiDev) [reply calligraphy/src/main/java/io/github/inflationx/calligraphy3/Calligraphy.java:29]: @lucalves I am not sure why you suggesting this change here. Javadoc has not changed and it's not part of this pull request. Won't change.

**Comment by davelogan** [comment]
When is this fix schedule for a release?

**Pull request closed without merging**